### PR TITLE
[trusted-types] Properly augment global to prevent conflicts with lib.dom.d.ts et al.

### PR DESCRIPTION
**Pull request by eps1lon**
This is the proper patern to augment global interfaces that are defined by the runtime (see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60837#pullrequestreview-1009631323)

The previous approach would break once `lib.dom.d.ts` declares these interfaces.

The previous approach also made it impossible for other libraries to use these types.
For example, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60691/ had to be reverted.
With proper module augmentation we can land https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60691/ again.

Verified that the new approach will allow landing https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60691/: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61166

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
   - https://developer.mozilla.org/en-US/docs/Web/API/TrustedHTML
   - https://developer.mozilla.org/en-US/docs/Web/API/TrustedScript
   - https://developer.mozilla.org/en-US/docs/Web/API/TrustedScriptURL
   - https://developer.mozilla.org/en-US/docs/Web/API/TrustedTypePolicy
   - https://developer.mozilla.org/en-US/docs/Web/API/TrustedTypePolicyFactory
- ~[ ]~ If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## trusted-types ([<kbd>unpkg</kbd>](https://unpkg.com/browse/trusted-types@latest/))
was missing the following properties:
1. TrustedTypePolicy
1. TrustedTypePolicyFactory
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0c07c86af4ab9365aad2b2bc620ee46e2821d7a9
</p>

File: types/trusted-types/index.d.ts [hunk 1]
```diff
@@ -4,30 +4,32 @@
...
 //                 Bjarki <https://github.com/bjarkler>
+//                 Sebastian Silbermann <https://github.com/eps1lon>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.1
 
 import * as lib from './lib';
 
 // Re-export the type definitions globally.
 declare global {
-    const TrustedHTML: typeof lib.TrustedHTML;
```
Comment (navya9singh) [inline types/trusted-types/index.d.ts:14]: This change would make `TrustedHTML` only a type and not a value. Is that what you intended to do? If not, then you can change `const TrustedHTML` to `var TrustedHTML` .

File: types/trusted-types/index.d.ts [hunk 2]
```diff
@@ -4,30 +4,32 @@
 //                 Damien Engels <https://github.com/engelsdamien>
 //                 Emanuel Tesar <https://github.com/siegrift>
 //                 Bjarki <https://github.com/bjarkler>
+//                 Sebastian Silbermann <https://github.com/eps1lon>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.1
```
Comment (bjarkler) [inline types/trusted-types/index.d.ts:9]: Since this is a backwards-incompatible change, please bump to a new major version.

Comment (RyanCavanaugh) [reply types/trusted-types/index.d.ts:9]: That is not how versioning on DefinitelyTyped works.

**Review by RyanCavanaugh — APPROVED** [review]

**Pull request closed without merging**